### PR TITLE
refactor: update compose and Makefile to use retail root env variable…

### PR DESCRIPTION
**Pull request by ejlee3**
… instead of hard coding for volume mounts

## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/retail-use-cases/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #**issue_number**

## Test Instructions if applicable

git clone this PR
make run docker compose for all use cases
ensure that the env variable changes have no impact on the make targets

### CODE REVIEW

File: use-cases/demos/Makefile [hunk 1]
```diff
@@ -10,36 +10,38 @@
...
 DEPENDENCY_COMPOSE ?= add_dependent_services.yml
+RETAIL_USE_CASE_ROOT ?= ../..
+RESULTS_DIR ?= ../../results
 
 build:
 	docker build --tag python-demo:dev -f Dockerfile .
 
 gen_classification:
-	@PIPELINE_COUNT=${PIPELINE_COUNT} docker compose -f $(DEPENDENCY_COMPOSE) -f add_demo_classification.yml convert > $(DOCKER_COMPOSE)
+	@PIPELINE_COUNT=${PIPELINE_COUNT} @RETAIL_USE_CASE_ROOT=$(RETAIL_USE_CASE_ROOT) @RESULTS_DIR=$(RESULTS_DIR) docker compose -f $(DEPENDENCY_COMPOSE) -f add_demo_classification.yml convert > $(DOCKER_COMPOSE)
```
Comment (jim-wang-intel) [inline use-cases/demos/Makefile:20]: remove @ symbol for REATIL_USE_CASE_ROOT and RESULTS_DIR as it is not working when I tested it

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
some syntax error when i ran make file targets from the project root directory: make run-gprc_python for example

File: use-cases/dlstreamer/Makefile [hunk 2]
```diff
@@ -13,7 +15,7 @@ build-realsense:
 	docker build --build-arg HTTPS_PROXY=${HTTPS_PROXY} --build-arg HTTP_PROXY=${HTTP_PROXY} --target build-realsense -t dlstreamer:realsense -f Dockerfile .
 
 run:
-	PIPELINE_SCRIPT=$(PIPELINE_SCRIPT) PIPELINE_COUNT=$(PIPELINE_COUNT) docker compose -f $(DOCKER_COMPOSE) up -d
+	PIPELINE_SCRIPT=$(PIPELINE_SCRIPT) PIPELINE_COUNT=$(PIPELINE_COUNT) @RETAIL_USE_CASE_ROOT=$(RETAIL_USE_CASE_ROOT) @RESULTS_DIR=$(RESULTS_DIR) docker compose -f $(DOCKER_COMPOSE) up -d
```
Comment (brian-intel) [inline use-cases/dlstreamer/Makefile:18]: remove @ from RETAIL_USE_CASE_ROOT & RESULTS_DIR

File: use-cases/grpc_go/Makefile [hunk 3]
```diff
@@ -5,12 +5,14 @@
...
 PIPELINE_COUNT ?= 1
+RETAIL_USE_CASE_ROOT ?= ../..
+RESULTS_DIR ?= ../../results
 
 build:
 	docker build --tag grpc_go:dev .
 
 run:
-	PIPELINE_COUNT=$(PIPELINE_COUNT) docker compose -f $(DOCKER_COMPOSE) up -d
+	PIPELINE_COUNT=$(PIPELINE_COUNT) @RETAIL_USE_CASE_ROOT=$(RETAIL_USE_CASE_ROOT) @RESULTS_DIR=$(RESULTS_DIR) docker compose -f $(DOCKER_COMPOSE) up -d
```
Comment (brian-intel) [inline use-cases/grpc_go/Makefile:15]: remove @ from RETAIL_USE_CASE_ROOT & RESULTS_DIR

File: use-cases/grpc_python/Makefile [hunk 4]
```diff
@@ -20,7 +22,7 @@ model-names:
 	@echo
 
 run:
-	PIPELINE_COUNT=$(PIPELINE_COUNT) MODEL_NAME=$(MODEL_NAME) docker compose -f $(DOCKER_COMPOSE) up -d
+	PIPELINE_COUNT=$(PIPELINE_COUNT) MODEL_NAME=$(MODEL_NAME) @RETAIL_USE_CASE_ROOT=$(RETAIL_USE_CASE_ROOT) @RESULTS_DIR=$(RESULTS_DIR) docker compose -f $(DOCKER_COMPOSE) up -d
```
Comment (brian-intel) [inline use-cases/grpc_python/Makefile:25]: remove @ from RETAIL_USE_CASE_ROOT & RESULTS_DIR

File: use-cases/gst_capi/Makefile [hunk 5]
```diff
@@ -62,34 +64,34 @@ build_capi_yolov5_ensemble:
 		-t $(OVMS_CPP_DOCKER_IMAGE)-capi-gst-ovms-capi_yolov5_ensemble:$(OVMS_CPP_IMAGE_TAG)
 
 gen_capi_yolov5:
-	@PIPELINE_COUNT=${PIPELINE_COUNT} docker compose -f $(CAMERA_SIMULATOR_COMPOSE) -f add_gst_capi_yolov5.yml convert > $(DOCKER_COMPOSE)
+	@PIPELINE_COUNT=${PIPELINE_COUNT} @RETAIL_USE_CASE_ROOT=$(RETAIL_USE_CASE_ROOT) @RESULTS_DIR=$(RESULTS_DIR) docker compose -f $(CAMERA_SIMULATOR_COMPOSE) -f add_gst_capi_yolov5.yml convert > $(DOCKER_COMPOSE)
```
Comment (brian-intel) [inline use-cases/gst_capi/Makefile:67]: remove @ from RETAIL_USE_CASE_ROOT & RESULTS_DIR

**Review by jim-wang-intel — COMMENTED** [review]
please rebase and get the smoke tests and run smoke tests from the root directory: `make run-smoke-tests` to see if works

**Review by brian-intel — APPROVED** [review]

File: use-cases/demos/Makefile [hunk 6]
```diff
@@ -10,36 +10,38 @@
...
-	@PIPELINE_COUNT=$(PIPELINE_COUNT) docker compose -f $(DOCKER_COMPOSE) up -d
+	@PIPELINE_COUNT=$(PIPELINE_COUNT) RETAIL_USE_CASE_ROOT=$(RETAIL_USE_CASE_ROOT) RESULTS_DIR=$(RESULTS_DIR) docker compose -f $(DOCKER_COMPOSE) up -d
 
 down_classification:
-	@PIPELINE_COUNT=$(PIPELINE_COUNT) docker compose -f add_demo_classification.yml down
+	@PIPELINE_COUNT=$(PIPELINE_COUNT) RETAIL_USE_CASE_ROOT=$(RETAIL_USE_CASE_ROOT) RESULTS_DIR=$(RESULTS_DIR) docker compose -f add_demo_classification.yml down
 
 gen_instance_segmentation:
-	@PIPELINE_COUNT=${PIPELINE_COUNT} docker compose -f $(DEPENDENCY_COMPOSE) -f add_demo_instance_segmentation.yml convert > $(DOCKER_COMPOSE)
+	@PIPELINE_COUNT=${PIPELINE_COUNT} RETAIL_USE_CASE_ROOT=$(RETAIL_USE_CASE_ROOT) docker compose -f $(DEPENDENCY_COMPOSE) -f add_demo_instance_segmentation.yml convert > $(DOCKER_COMPOSE)
```
Comment (jim-wang-intel) [inline use-cases/demos/Makefile:29]: missing RESULT_DIR="$(RESULT_DIR)"

**Review by jim-wang-intel — APPROVED** [review]
tested locally and works perfect for me!

**Pull request merged**